src/hotspot/share/classfile/classLoader.cpp

```diff
@@ -826,11 +827,14 @@ bool ClassLoader::contains_append_entry(const char* name) {
   return false;
 }
 
+// The boot append entries are added with a lock, and read lock free.
 void ClassLoader::add_to_boot_append_entries(ClassPathEntry *new_entry) {
   if (new_entry != NULL) {
+    MutexLocker ml(Bootclasspath_lock, Mutex::_no_safepoint_check_flag);
     if (_last_append_entry == NULL) {
-      assert(_first_append_entry == NULL, "boot loader's append class path entry list not empty");
-      _first_append_entry = _last_append_entry = new_entry;
+      _last_append_entry = new_entry;
+      assert(first_append_entry() == NULL, "boot loader's append class path entry list not empty");
+      Atomic::store(&_first_append_entry_list, new_entry);
```

Comment:
This should probably be `OrderAccess::release_store`.

Reply:
Ok, thanks.